### ISSUE TITLE
Mac Issue with Mysql...

### ISSUE THREAD

**Issue by marcodeluca** [laravel/valet#917]
Hey, so this has nothing to do specifically with valet, but I'll post here on the off chance  @drbyte or someone has a suggestion. It's a relatively new issue that I can't find a solution to. After installing valet, here is the mysql error:

```
dyld: Library not loaded: /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib
Referenced from: /usr/local/bin/mysql
Reason: image not found
```

**openssl version**
OpenSSL 1.1.1d  10 Sep 2019

but uninstalling and re-installing openssl: **brew link openssl**

`Warning: Refusing to link macOS-provided software: openssl@1.1`

As far as I can tell, it's related to apple's system integrity protection (SIP).

Aside from some other issues that were resolved yesterday, PHP and NGINX were installed by valet just fine. So I just need to figure out a way to get MySQL working. Any help would be appreciated, even though I dont think any of this is related to valet.

**Comment by drbyte** [comment]
The message tells me that the mysql binary is expecting to load the lib for libcrypto, but is errantly expecting an older version.

Homebrew installs two different openssl versions depending on certain dependency criteria: 1.0.0 and 1.1.1

I'm guessing when you installed MySql you had the older 1.0.0 openssl installed at the time, but after the reinstall of other things now you're left with the newer one.

I'm similarly guessing that if you uninstall mysql and re-install it, hopefully it will link things up properly.

There are a half dozen other Issue threads here related to "dyld Library not loaded" and "libcrypto" version mismatches. You might find some keys in there, but I'm pretty sure in the end it really boils down to just uninstalling the old incompatible pieces and updating with new.

**Comment by marcodeluca** [comment]
One would think so, but that was one of the first things I did. I don't have patience for anymore apple issues right now, but I appreciate the suggestion. Maybe it's one of those sleep on it, restart and try again things.

**Comment by drbyte** [comment]
Given that your mysql is looking for 1.0.0 openssl files, I suppose you might get it to work by getting homebrew to install the older 1.0.0 openssl alongside the more current 1.1.1.

**Comment by drbyte** [comment]
@marcodeluca I'm pretty sure that you can fix that mysql/openssl mismatch be reinstalling mysql. (Perhaps removing it first, and then installing it "fresh").

Here's why: it seems the mysql "recipe" is from time to time bound to a specific openssl version, and I'm guessing yours was bound to the 1.0 version at the time it was installed. But now with all the various updates we did to fix the other issue discussed, that 1.0 is no longer installed, but your older un-updated mysql is still expecting the 1.0 files because it was compiled to depend on them. 
Even the [mysql recipe has internal notes about clashes with version numbers](https://github.com/Homebrew/homebrew-core/blob/47a6f2d5cc3c1cac9d5387d19ca926193d7095d3/Formula/mysql.rb#L55-L76).

Thus my recommendation to remove and reinstall your mysql version.

**Comment by marcodeluca** [comment]
Hey, yah sorry I forgot to come back to this - after a restart, uninstall and re-install it was back up and running. Thanks again @drbyte
